kotlinx-coroutines-core/common/src/channels/Channels.common.kt

```diff
@@ -189,11 +200,68 @@ public suspend inline fun <E> ReceiveChannel<E>.consumeEach(action: (E) -> Unit)
  * ```
  */
 public suspend fun <E> ReceiveChannel<E>.toList(): List<E> = buildList {
```

Comment:
We forgot about lost elements (`onUndeliveredElement`). It should be at least mentioned in the doc, and possibly `onUndeliveredElement` could be added as a parameter.

`onUndeliveredElement` overcomplicates the signature for no real use-case, but provides completeness.